lib/Wikia/src/Domain/User/Preferences/LocalPreference.php

```diff
@@ -14,6 +14,12 @@ function __construct( $name, $value, $wikiId ) {
 		Assert::true( !empty( $name ), "invalid preference name" );
 		Assert::true( !empty( $wikiId ), "invalid wiki id" );
 
+		if ( $value === "true" ) {
+			$value = true;
+		} elseif ( $value === "false" ) {
+			$value = false;
+		}
```

Comment:
I think that would turn everything into a boolean, right? But that's not what we want.

```
[14:20] ~/Downloads> cat test.php
<?php
var_dump(filter_var("true", FILTER_VALIDATE_BOOLEAN));
var_dump(filter_var("false", FILTER_VALIDATE_BOOLEAN));
var_dump(filter_var("1", FILTER_VALIDATE_BOOLEAN));
var_dump(filter_var("0", FILTER_VALIDATE_BOOLEAN));
var_dump(filter_var(1, FILTER_VALIDATE_BOOLEAN));
var_dump(filter_var(0, FILTER_VALIDATE_BOOLEAN));
var_dump(filter_var("something-else", FILTER_VALIDATE_BOOLEAN));
var_dump(filter_var("42", FILTER_VALIDATE_BOOLEAN));
var_dump(filter_var(42, FILTER_VALIDATE_BOOLEAN));
[14:20] ~/Downloads> php test.php
bool(true)
bool(false)
bool(true)
bool(false)
bool(true)
bool(false)
bool(false)
bool(false)
bool(false)
```